### PR TITLE
Code refactoring for 0.3.0

### PR DESCRIPTION
**Pull request by chfw**
- "溶解"了template.py
- "溶解"了class Helpers
- 其他小改动

### CODE REVIEW

File: pyecharts/conf.py [hunk 1]
```diff
@@ -61,3 +83,133 @@ def convert_jshost_string(jshost):
         if jshost[-1:] in ('/', '\\'):
             jshost = jshost[:-1]
         return jshost
+
+    def get_js_library(self, pinyin):
+        return DEFAULT_JS_LIBRARIES.get(pinyin, pinyin)
```
Comment (kinegratii) [inline pyecharts/conf.py]: Yeah, so users should not directly access `DEFAULT_JS_LIBRARIES` and `CITY_NAME_PINYIN_MAP `

File: pyecharts/conf.py [hunk 2]
```diff
@@ -61,3 +83,133 @@ def convert_jshost_string(jshost):
...
+        :param dependencies:
+        :param jshost:
+        :return:
+        """
+        _d = _ensure_echarts_is_in_the_front(dependencies)
+        # if no nick name register, we treat the location as location.js
+        require_conf_items = [
+            "'%s': '%s/%s'" % (key,
+                               jshost,
+                               CURRENT_CONFIG.get_js_library(key))
```
Comment (kinegratii) [inline pyecharts/conf.py]: 在一个类 `PyEchartsConfig` 的  staticmethod 方法引用了该类的一个全局实例(CURRENT_CONFIG)，将 `produce_require_configuration` 改实例方法更为合适。

File: pyecharts/base.py [hunk 3]
```diff
@@ -61,31 +62,27 @@ def render_embed(self):
...
         """
-        embed = 'chart_component.html'
-        default_engine = template.create_builtin_template_engine()
-        tmp = default_engine.get_template(embed)
         my_option = utils.json_dumps(self._option, indent=4)
-        html = tmp.render(my_option=my_option,
-                          chart_id=self._chart_id,
-                          my_width=self.width,
-                          my_height=self.height)
+        html = engine.render('chart_component.html',
```
Comment (chenjiandongx) [inline pyecharts/base.py]: 这个不错，把 `render` 函数都给统一起来了。

File: pyecharts/conf.py (hunk 2 above)
Comment (chenjiandongx) [reply pyecharts/conf.py]: 他可能是看 self, 参数没用到，所以就改为 静态函数 了

File: pyecharts/base.py [hunk 4]
```diff
@@ -128,25 +125,29 @@ def _repr_html_(self):
...
         """
         my_option = utils.json_dumps(self._option, indent=4)
-        return template.render("notebook_chart_component.html",
-                               my_option=my_option,
-                               chart_id=self._chart_id)
+        return engine.render("notebook_chart_component.html",
+                             my_option=my_option,
+                             chart_id=self._chart_id)
+
+    def _add_chinese_map(self, map_name_in_chinese):
```
Comment (chenjiandongx) [inline pyecharts/base.py]: 👍

File: test.sh [hunk 5]
```diff
@@ -1,2 +1,2 @@
 cd test
-nosetests --with-coverage --cover-package pyecharts --cover-package test && flake8 pyecharts --builtins=unicode,xrange,long
+nosetests --with-coverage --cover-package pyecharts --cover-package test && cd .. && flake8 --exclude test,docs --builtins=unicode,xrange,long
```
Comment (chenjiandongx) [inline test.sh]: test 的不要加入吗

**Comment by chenjiandongx** [comment]
整体上来说没什么大问题，因为这个 PR 主要是对代码进行了整理，没涉及功能变动。

File: test.sh (hunk 5 above)
Comment (chfw) [reply test.sh]: 我认为是需要。不过下一PR做这个更好。主题明确 好review

File: pyecharts/conf.py (hunk 1 above)
Comment (chfw) [reply pyecharts/conf.py]: 嗯

File: pyecharts/conf.py (hunk 2 above)
Comment (chfw) [reply pyecharts/conf.py]: 是的 没有self 就变static了

**Comment by chfw** [comment]
需要你们点击approve我才能merge. 现在 我们已经正规化了.
